src/mod/server.mod/server.c

```diff
@@ -1684,6 +1687,8 @@ static void server_secondly()
   deq_msg();
   if (!resolvserv && serv < 0)
     connect_server();
+  if (sasl_timeout_time == now)
```

Comment:
not robust against us missing a second (clock adjustment or blocking execution of anything), better check >= here, but see the other comment for alternative approaches

Reply:
yes. fixed.